usl_pipeline/cloud_functions/main.py

```diff
@@ -903,3 +900,98 @@ def _write_as_npy(blob: storage.Blob, array: NDArray) -> None:
     npy_file.flush()
     npy_file.seek(0)
     blob.upload_from_file(npy_file)
+
+
+def _start_feature_rescaling_if_ready(feature_bucket: storage.Bucket, blob_path: str):
+    """Creates scaled version for every chunk file in the storage bucket parent folder.
+
+    Output files with scaled feature matrices will be placed to the same folder where
+    unscaled files (following "chunk_*" name pattern) are stored but will get "scaled_"
+    prefix. Rescaling only starts when metadata of study area (corresponding to the
+    parent folder name) contains chunks counters giving exactly the same number of
+    chunks as are currently stored in the bucket folder. This should happen when the
+    last feature chunk is stored.
+
+    Args:
+        feature_bucket: Storage bucket with feature matrices.
+        blob_path: Path pointing to one of unscaled feature matrix files with the name
+            following the "chunk_*" pattern. Other files are ignored.
+    """
+    study_area_name, chunk_name = _parse_chunk_path(blob_path)
+    # Let's look up chunk metadata and check if scaling is needed
+    db = firestore.Client()
+    chunk_metadata = metastore.StudyAreaSpatialChunk.get_if_exists(
+        db, study_area_name, chunk_name
+    )
+    if chunk_metadata is None:
+        logging.info(f"Chunk metadata was not found for {study_area_name}/{chunk_name}")
+        return
```

Comment:
should this be an error condition? should we raise something here?

Reply:
No, this may happen for scaled files for instance (these files have names completely unrelated to chunk_id in metadata. So this is expected case. I just wanted to see in logs that everything is going as expected.

Reply:
sounds good -- might add a comment explaining this